thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/business/service/TermSetupService.java

```diff
@@ -96,14 +101,14 @@ private List<WordFrequency> getItemsTaken(List<WordFrequency> termsFromApi) {
         return termsFromApi.subList(0, itemsTaken);
     }
 
-    private List<WordFrequency> getTermsFromApi(Category category) {
+    private List<WordFrequency> getTermsFromApi(SearchPhrase searchPhrase) {
         List<WordFrequency> termsFromApi = null;
         try {
             termsFromApi = datamuseClient.meansLike(
-                            category.getSearchWord(),
+                            searchPhrase.getSearchWord(),
                             Map.of(
                                     DatamuseParam.Code.MD, META_FLAG_F,
-                                    DatamuseParam.Code.MAX, "1000"
+                                    DatamuseParam.Code.MAX, searchPhrase.getNumberOfRecords().toString()
```

Comment:
I just noticed this by accident, we should download more words from dataMuse because we are rejecting some of them but we would like to have `numberOfRecords` of them in the database in the end. So better download some more and as an additional step, reject some of them just before saving them in database (i.e. method `getNewTermsFromApiTaken`)

Reply:
You really think that is a good idea? we don't really know how much more to download. I agree it should be changed I was mentioning this earlier that every try we are trying to download new words from same list of words and we can't get to satisfy this condition:
```
        if (numberOfTermsBySearchPhrase >= numberOfRecords) {
            return;
        }
```

Reply:
We can just check the
```
        if (numberOfTermsBySearchPhrase >= 0.8*numberOfRecords) {
            return;
        }
```
but I understand it would be better to have as many records as variable numberOfRecords suggesting